packages/xforms-engine/src/client/ModelValueNode.ts

```diff
@@ -0,0 +1,40 @@
+import type { ValueNodeDefinition } from '../model/ValueNodeDefinition.ts';
+import type { BaseNode, BaseNodeState } from './BaseNode.ts';
+import type { GeneralParentNode } from './hierarchy.ts';
+import type { RootNode } from './RootNode.ts';
+import type { LeafNodeValidationState } from './validation.ts';
+
+export interface ModelValueNodeState extends BaseNodeState {
+	get label(): null;
+	get hint(): null;
+	get children(): null;
+	get valueOptions(): null;
```

Comment:
I'm not sure! I think we might be in new territory having a concept of distinct model-only nodes. I did a quick spike to see if any `scenario` tests would fail, expecting _at least_ one or two failures on `relevant` checks, but I don't see any.

Keep in mind that there's already another (mostly[^1]) model-only node type: `SubtreeNode`. We don't presently make a special case for these state properties on `SubtreeNode.currentState`, and if we were to do that here, I'd expect to do it there too.

I think it's also worth reconsidering whether model-only nodes of any type are even pertinent to a client. More precisely, whether they're pertinent to **every client**. I did another quick spike to jog my memory, and there are definitely several `scenario` tests which fail if we filter them out[^2].

But I definitely think it's an open question whether they even provide any value to the `web-forms` Vue client. I think that question would have a similar answer for _some_ other presentational client use cases, depending on their use case: a hypothetical native app for form filling would line up with `web-forms`, but tooling for form design or debugging would almost certainly need a way to access model-only form aspects[^3].

Right now, my instinct on this one is to defer all of these questions, and follow up on them with a more thoughtful design around how we handle model-only aspects of form state at the engine/client boundary overall. I think it would be worthwhile to consider different "views" for different client use cases.

[^1]: I suspect there are edge cases where we designate a node a `SubtreeNode` even if it ultimately has descendants which aren't model-only. Probably worth filing an issue to confirm, and decide whether that's right.

[^2]: It's worth mentioning that this made me doubt that some of the affected tests are "really integration tests". I tend to think about model-only form features as basically "implementation details" of the form, at least while filling an instance. A stricter conception of "integration test" would probably only test their submission state.

[^3]: Tooling use cases would also almost certainly have more reason to access to `definition` aspects of a form. Now that I'm giving all of this more thought, that's something I want to keep in mind as I otherwise try to phase those out for our main user-facing client use case.